packages/xo-server/src/api/xostor.mjs

```diff
@@ -105,10 +106,45 @@ formatDisks.resolve = {
   host: ['host', 'host', 'administrate'],
 }
 
-export async function create({ description, disksByHost, force, ignoreFileSystems, name, provisioning, replication }) {
+export const create = defer(async function (
+  $defer,
+  { description, disksByHost, force, ignoreFileSystems, name, provisioning, replication }
+) {
   const hostIds = Object.keys(disksByHost)
-  const hosts = hostIds.map(hostId => this.getObject(hostId, 'host'))
 
+  const tmpBoundObjectId = `tmp_${hostIds.join(',')}_${Math.random().toString(32).slice(2)}`
+
+  const xostorLicenses = await this.getLicenses({ productType: 'xostor' })
+
+  const unboundLicenses = xostorLicenses.filter(license => license.boundObjectId === undefined)
+  let license = unboundLicenses.find(license => license.productId === 'xostor')
+
+  if (license === undefined) {
+    license = unboundLicenses.find(license => license.productId === 'xostor.trial')
```

Comment:
Shouldn't you check if it's bound and expired here as well?

Reply:
Yes, i will also add the expiration check here.